fieldpath/path.go

```diff
@@ -93,10 +93,10 @@ func MakePath(parts ...interface{}) (Path, error) {
 				return nil, fmt.Errorf("associative list key type path elements must have at least one key (got zero)")
 			}
 			fp = append(fp, PathElement{Key: t})
-		case value.Value:
+		case *value.Value:
```

Comment:
It doesn't seem good to want a pointer to an interface here. Why do we want that?